### PR TITLE
fix raw_coverage to have complete file list, add file_counts for regu…

### PR DESCRIPTION
**Pull request by jmaher**
…lar reporting

adr file_counts:
┌──────────────┬─────────────┬───────────────┐
│ Revision     │ Total Files │ Total Covered │
├──────────────┼─────────────┼───────────────┤
│ ed94dc665071 │ 90236       │ 231245        │
│ 41a66272bd11 │ 86661       │ 231245        │
│ 277e2ae05f74 │ 98077       │ 231245        │
│ cbebb4f65eea │ 88749       │ 231241        │
│ 6409f04e33d6 │ 93420       │ 231237        │
│ 2bdf6eed0f64 │ 84760       │ 231237        │
│ 05ae89bebb64 │ 83436       │ -1            │
│ 2203d23429f6 │ 97376       │ 231227        │
│ d16b52f5d195 │ 93991       │ 231227        │
│ bda35e3d8ce4 │ 87273       │ 231211        │
└──────────────┴─────────────┴───────────────┘

in addition, the code for querying all files covered and related suites accounts for large subdirs and will iterate to subdirs if needed- still sort of sloppy, but is a lot more accurate than we had prior to this PR

### CODE REVIEW

**Comment by jmaher** [comment]
@ahal can you review the python
@klahnakoski can you review the .query files?

File: adr/recipes/file_counts.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    for item in total_files:
+        if item[0] in by_revision:
+            by_revision[item[0]]['total'] = item[2]
+
+    data = []
+    dates = by_date.keys()
+    dates.sort(reverse=True)
+
+    # For a simple report 10 data points is good enough
+    for date in dates[0:10]:
```
Comment (ahal) [inline adr/recipes/file_counts.py]: You might want to add a --limit (see the task_durations recipe for an example) that defaults to 10. This way you can get a bigger sample if desired.

File: adr/recipes/raw_coverage.py [hunk 2]
```diff
@@ -42,14 +42,83 @@ def run(args):
...
+
+            if l in all_dirs:
+                continue
+
+            # check if the parent is already in the dir list
+            if '/'.join(l.split('/')[0:-1]) in all_dirs:
+                continue
+
+            if count < 50000:
+                print("adding %s to all_dirs: %s" % (l, count))
```
Comment (ahal) [inline adr/recipes/raw_coverage.py]: At the top of your file, add:

    log = logging.getLogger('adr')

Then turn these into:

    log.debug('adding ...')

This way, when running with `adr --quiet`, these messages won't get printed. This is important if we ever want to upload this to the gist.

**Comment by jmaher** [comment]
@ahal I have made the changes you recommended and they look to be working- thanks for the suggestions- feel free to merge this in if you want

**Pull request merged**